packages/sui-gateway/src/lib.rs

```diff
@@ -1,3 +1,184 @@
-pub mod base_types;
+use std::str::FromStr;
+
+use cosmwasm_std::Uint256;
+use error_stack::{report, Report, ResultExt};
+use multisig::key::PublicKey;
+use multisig::msg::SignerWithSig;
+use multisig::verifier_set::VerifierSet;
+use serde::{Deserialize, Serialize};
+use sha3::{Digest, Keccak256};
+use sui_types::SuiAddress;
+
 pub mod error;
-pub mod gateway;
+pub mod events;
+
+use error::Error;
+
+#[repr(u8)]
```

Comment:
Move this to a different file, say `types.rs`? I don't like making `lib.rs` files complex, instead of keeping their focus on module inclusions, visibility restriction/re-exports.

Reply:
It's not a random collection. The crate is `sui-gateway`, and these are the Sui gateway type definitions, so `sui-gateway::types` makes sense to me. We can make it more verbose like `gateway-types.rs` if you like (could also break it out into message.rs and proof.rs type definitions). But this crate already has a few modules, so don't see the point of bloating lib.rs

Reply:
I agree with Chris. I don't like having mods in a crate before it needs the separation.